src/Events/Services/AppEventsService.cs

```diff
@@ -48,17 +48,45 @@ public class AppEventsService : IAppEventsService
         }
 
         /// <inheritdoc/>
-        public async Task<string> StoreCloudEvent(CloudEvent cloudEvent)
+        public async Task<string> SaveToDatabase(CloudEvent cloudEvent)
+        {
+            try
+            {
+                cloudEvent = await _repository.Create(cloudEvent);
```

Comment:
C#/.Net question: Create overwrites cloudEvent. If it fails, ex: casts exception during deserialization of the cloud event coming from db, will the variable be overwritten by something that makes cloudEvent.Id potentially fail? Or is this safe behaviour when working with async functions in .Net?